homeassistant/components/verisure/__init__.py

```diff
@@ -48,9 +48,19 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     # Set up all platforms for this device/entry.
     await hass.config_entries.async_forward_entry_setups(entry, PLATFORMS)
 
+    # Update options
+    entry.async_on_unload(entry.add_update_listener(update_listener))
+
     return True
 
 
+async def update_listener(hass: HomeAssistant, entry: ConfigEntry):
+    """Handle options update."""
+    # Propagate configuration change.
```

Comment:
I didn't check but typically we just reload the config entry to load new settings.

Reply:
Yes, that also works here, but that is a much heavier operation, including triggering an extra call to fetch data from Verisure, which are (as I understand it) rate limiting their API quite strict. In this way, we avoid that extra external request when only updating local configuration. I checked a few different integrations, and some reloads, while others do not. Perhaps I am suboptimisng here?

(What I actually found  a bit strange, was that the lock entity fetches `code_format` on update, and not when opening the options flow in the UI.)

Reply:
Not sure I understand the last sentence?

Reply:
> Not sure I understand the last sentence?

When debugging, I noticed that `code_format()` was called about once a minute, apparently after data is received from Verisure.

Reply:
That's not important for this PR though. I am happy to change to reload the config entry if that's the preferred way.

Reply:
No, I'm now wondering if the `update_listener` even have to do anything since `code_format` is only used for the services and frontend which I think is reading live.
You ticked the box but have you tested this?

Reply:
Yes, of course I tested this :)

Without `update_listener`, the code_format in the OptionsFlow is not updated until after the next data fetching. Here is with added logging in `code_format()` in `verisure/lock.py`:

```
2023-08-29 21:51:03.063 DEBUG (MainThread) [homeassistant.components.verisure] Finished fetching verisure data in 0.545 seconds (success: True)
2023-08-29 21:51:03.063 DEBUG (MainThread) [homeassistant.components.verisure] code_format: ^\d{4}$
```
_Changing number of digits in Verisure options UI here_
_No change yet in Lock options UI_
```
2023-08-29 21:52:03.092 DEBUG (MainThread) [homeassistant.components.verisure] Finished fetching verisure data in 0.572 seconds (success: True)
2023-08-29 21:52:03.092 DEBUG (MainThread) [homeassistant.components.verisure] code_format: ^\d{6}$
```
_Change effective in Lock options UI_

Reply:
> `code_format` is only used for the services and frontend which I think is reading live.

This is the behaviour I expected as well, but apparently that's not the case!